collections/sum_of.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { Selector } from "./types.ts";
+
+/**
+ * Applies the given selector to all elements in the given collection and calculates the sum of the results
+ *
+ * Example:
+ *
+ * ```typescript
+ * import { sumOf } from "./sum_of.ts"
+ *
+ * const people = [
+ *     { name: 'Anna', age: 34 },
+ *     { name: 'Kim', age: 42 },
+ *     { name: 'John', age: 23 },
+ * ]
+ * const totalAge = sumOf(people, i => i.age)
+ *
+ * console.assert(totalAge === 99)
```

Comment:
We should start fixing that equality example thing. Others have used `assertEquals` now and I think that is the best for the moment:

```suggestion
 * import { sumOf } from "./sum_of.ts"
 * import { assertEquals } from "../testing/mod.ts"
 *
 * const people = [
 *     { name: 'Anna', age: 34 },
 *     { name: 'Kim', age: 42 },
 *     { name: 'John', age: 23 },
 * ]
 * const totalAge = sumOf(people, i => i.age)
 *
 * assertEquals(totalAge, 99)
```